yarn-project/acir-simulator/src/acvm/oracle/oracle.ts

```diff
@@ -80,35 +80,32 @@ export class Oracle {
       +offset,
     );
 
-    const contractAddress = notes[0]?.contractAddress ?? Fr.ZERO;
+    const preimageLength = notes?.[0]?.preimage.length ?? 0;
+    if (!notes.every(({ preimage }) => preimageLength === preimage.length)) {
+      throw new Error('Preimages for a particular note type should all be the same length.');
+    }
 
-    const isSome = new Fr(1); // Boolean. Indicates whether the Noir Option<Note>::is_some();
-    const realNotePreimages = notes.flatMap(({ nonce, preimage }) => [nonce, isSome, ...preimage]);
-    const preimageLength = notes[0]?.preimage.length ?? 0;
-    const returnHeaderLength = 2; // is for the header values: `notes.length` and `contractAddress`.
-    const extraPreimageLength = 2; // is for the nonce and isSome fields.
-    const extendedPreimageLength = preimageLength + extraPreimageLength;
-    const numRealNotes = notes.length;
-    const numReturnNotes = Math.floor((+returnSize - returnHeaderLength) / extendedPreimageLength);
-    const numDummyNotes = numReturnNotes - numRealNotes;
-
-    const dummyNotePreimage = Array(extendedPreimageLength).fill(Fr.ZERO);
-    const dummyNotePreimages = Array(numDummyNotes)
-      .fill(dummyNotePreimage)
-      .flatMap(note => note);
-
-    const paddedZeros = Array(
-      Math.max(0, +returnSize - returnHeaderLength - realNotePreimages.length - dummyNotePreimages.length),
-    ).fill(Fr.ZERO);
-
-    return [notes.length, contractAddress, ...realNotePreimages, ...dummyNotePreimages, ...paddedZeros].map(v =>
-      toACVMField(v),
-    );
-  }
+    const contractAddress = notes[0]?.contractAddress ?? Fr.ZERO;
 
-  async checkNoteHashExists([nonce]: ACVMField[], [innerNoteHash]: ACVMField[]): Promise<ACVMField> {
-    const exists = await this.typedOracle.checkNoteHashExists(fromACVMField(nonce), fromACVMField(innerNoteHash));
-    return toACVMField(exists);
+    // Values indicates whether the note is settled or transient.
+    const noteTypes = {
```

Comment:
Replace `is_some` with the note type. `is_some` is not needed. We can learn about if a note is real by checking the first field in the return data, which is the number of real notes.